### PR TITLE
LPS-152226 Deprecate `disableToggleBoxes` and replace its usages

### PR DESCRIPTION
**Pull request by kresimir-coko**
After careful investigation I've come to the conclusion that this utility was a more specific variant of our [toggleDisabled](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/toggle_disabled.js) util. I've changed usages of this util to reflect that conclusion.

## 🧪 Testing

I've tested only one of three usages, I couldn't find the other 2 usages. I've tested the one inside `edit_feed.jsp` that can be tested by enabling Feeds first: System settings -> Web Content -> Administration -> Show feeds. Then go to Web Content > Feeds and create a new Feed. 

### 😕 Confusion

The given example above has the elements in question set as hidden, so the functionality should be able to be removed in that example, but I wanted to play it safe and keep everything as it was. Due to this, in order to test it, you will need to replace the `type="hidden"` with `type="text"` and `type="checkbox"`, respectively ([in this part](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_feed.jsp#L149-L150)).

### CODE REVIEW

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/basic_info.jsp [hunk 1]
```diff
@@ -101,11 +101,20 @@ DDMStructure ddmStructure = journalEditArticleDisplayContext.getDDMStructure();
...
-				true
+			var autoArticleInput = document.getElementById(
+				'<portlet:namespace />autoArticleId'
 			);
+			var newArticleInput = document.getElementById(
+				'<portlet:namespace />newArticleId'
+			);
+
+			if (autoArticleInput && newArticleInput) {
+				newArticleInput.disabled = true && autoArticleInput.checked;
```
Comment (bryceosterhaus) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/basic_info.jsp:112]: Hmm, this line is a bit odd. Why is there an explicit `true` value here?

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_feed.jsp [hunk 2]
```diff
@@ -411,11 +411,16 @@ renderResponse.setTitle((feed == null) ? LanguageUtil.get(request, "new-feed") :
...
-	Liferay.Util.disableToggleBoxes(
-		'<portlet:namespace />autoFeedId',
-		'<portlet:namespace />newFeedId',
-		true
-	);
+	var autoFeedInput = document.getElementById('<portlet:namespace />autoFeedId');
+	var newFeedCheckbox = document.getElementById('<portlet:namespace />newFeedId');
+
+	if (autoFeedInput && newFeedCheckbox) {
+		newFeedCheckbox.disabled = true && autoFeedInput.checked;
```
Comment (bryceosterhaus) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/edit_feed.jsp:418]: Same here

File: modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_public_render_parameters.jsp [hunk 3]
```diff
@@ -23,6 +23,7 @@ List<PublicRenderParameterConfiguration> publicRenderParameterConfigurations = (
 Set<PublicRenderParameter> publicRenderParameters = (Set<PublicRenderParameter>)request.getAttribute(WebKeys.PUBLIC_RENDER_PARAMETERS);
 %>
 
+<h1>edit_public_render_parameters</h1>
```
Comment (bryceosterhaus) [inline modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_public_render_parameters.jsp:26]: Is this supposed to be here?

File: modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_public_render_parameters.jsp [hunk 4]
```diff
@@ -140,11 +141,20 @@ Set<PublicRenderParameter> publicRenderParameters = (Set<PublicRenderParameter>)
...
-			true
+		var ignoreInput = document.getElementById(
+			'<portlet:namespace /><%= PublicRenderParameterConfiguration.IGNORE_PREFIX + HtmlUtil.escapeJS(publicRenderParameterConfiguration.getPublicRenderParameterName()) %>'
 		);
+		var mappingInput = document.getElementById(
+			'<portlet:namespace /><%= PublicRenderParameterConfiguration.MAPPING_PREFIX + HtmlUtil.escapeJS(publicRenderParameterConfiguration.getPublicRenderParameterName()) %>'
+		);
+
+		if (ignoreInput && mappingInput) {
+			mappingInput.disabled = true && ignoreInput.checked;
```
Comment (bryceosterhaus) [inline modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_public_render_parameters.jsp:152]: same as above

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/basic_info.jsp (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/article/basic_info.jsp:112]: Well, sure, now that I look at it after not staring at it for hours it doesn't make sense. But I was basically trying to replicate the old implementation. But yea, looks dumb now that I see it again haha.

**Comment by kresimir-coko** [comment]
@bryceosterhaus I've rebased and fixed your concerns, pushing the new commit

**Comment by kresimir-coko** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:forward

**Pull request closed without merging**